### PR TITLE
[Cppcheck] Enable Sider's recommended set of rules and add `suppressions-list` option

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Enable Sider's recommended set of rules.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

https://github.com/sider/sideci/issues/10671

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

**Comment by sleeeto** [comment]
@sleeeto Can you check my changes of methods, variables in cppcheck.rb ?

File: Gemfile.lock [hunk 1]
```diff
@@ -78,10 +78,6 @@ GEM
     nokogiri (1.11.7)
       mini_portile2 (~> 2.5.0)
       racc (~> 1.4)
-    nokogiri (1.11.7-x86_64-darwin)
-      racc (~> 1.4)
-    nokogiri (1.11.7-x86_64-linux)
-      racc (~> 1.4)
```
Comment (suetsuguk) [inline Gemfile.lock:84]: Restore this deleted block.

File: lib/runners/processor/cppcheck.rb [hunk 2]
```diff
@@ -50,6 +52,7 @@ def self.config_example
         addon: [cert, misra]
         bug-hunting: true
         parallel: false
+        suppressions-list: suppresions.txt
```
Comment (suetsuguk) [inline lib/runners/processor/cppcheck.rb:55]: typo.

```suggestion
        suppressions-list: suppressions.txt
```

File: lib/runners/processor/cppcheck.rb [hunk 3]
```diff
@@ -20,6 +20,7 @@ class NoTargetFiles < SystemError; end
         language: string?,
         'bug-hunting': boolean?,
         parallel: boolean?,
+        'suppressions-list': string?,
```
Comment (suetsuguk) [inline lib/runners/processor/cppcheck.rb:23]: Add a new smoke test for `suppressions-list` specified in sider.yml.

File: lib/runners/processor/cppcheck.rb [hunk 4]
```diff
@@ -111,6 +114,10 @@ def language
       config_linter[:language].then { |lang| lang ? ["--language=#{lang}"] : [] }
     end
 
+    def suppressions_list
+      (config_linter[:'suppressions-list'] || DEFAULT_SUPPRESSIONS_PATH).then { |file| (current_dir / file).exist? ? ["--suppressions-list=#{file}"] : [] }
```
Comment (suetsuguk) [inline lib/runners/processor/cppcheck.rb:118]:
```suggestion
      (config_linter[:'suppressions-list'] || DEFAULT_SUPPRESSIONS_PATH).then { |file| ["--suppressions-list=#{file}"] }
```

The existence check is not needed for this case.

File: lib/runners/processor/cppcheck.rb (hunk 3 above)
Comment (suetsuguk) [reply lib/runners/processor/cppcheck.rb:23]: For the new test, can you place the list file somewhere in a sub-directory (not in the repository root)?

File: Gemfile.lock (hunk 1 above)
Comment (sleeeto) [reply Gemfile.lock:84]: OK, I restored it.

File: lib/runners/processor/cppcheck.rb (hunk 2 above)
Comment (sleeeto) [reply lib/runners/processor/cppcheck.rb:55]: Thanks, I updated it.

File: lib/runners/processor/cppcheck.rb (hunk 3 above)
Comment (sleeeto) [reply lib/runners/processor/cppcheck.rb:23]: OK, I added a test.

File: lib/runners/processor/cppcheck.rb (hunk 4 above)
Comment (sleeeto) [reply lib/runners/processor/cppcheck.rb]: OK, I removed it.

**Comment by sleeeto** [comment]
@ybiquitous I have a question.
Where do I fix this test error below ?
`ConfigGeneratorTest#test_generate_with_tools` [/home/runner/work/runners/runners/test/config_generator_test.rb:28]:`

**Comment by suetsuguk** [comment]
I think you need to update this file https://github.com/sider/runners/blob/master/test/data/test_generate_with_tools.yml

**Comment by sleeeto** [comment]
@suetsuguk Thanks, ConfigGeneratorTest is passed.

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,6 +23,7 @@ Misc:
 
 - **PHPMD** Provide new recommended configuration [#2473](https://github.com/sider/runners/pull/2473)
 - Add exit status constants [#2492](https://github.com/sider/runners/pull/2492)
+- **Cppcheck** Enable new recommended configuration [#2474](https://github.com/sider/runners/pull/2474)
```
Comment (ybiquitous) [inline CHANGELOG.md:26]: [suggestion] How about this?

```suggestion
- **Cppcheck** Enable new recommended configuration and add `suppressions-list` option [#2474](https://github.com/sider/runners/pull/2474)
```

Comment (sleeeto) [reply CHANGELOG.md]: Thanks, I fixed it.

**Review by suetsuguk — APPROVED** [review]
LGTM. Thank you!

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,10 @@ All notable changes to this project will be documented in this file.
 
 [Full diff](https://github.com/sider/runners/compare/0.51.2...HEAD)
 
+Misc:
+
+- **Cppcheck** Enable new recommended configuration and add `suppressions-list` option  [#2474](https://github.com/sider/runners/pull/2474)
```
Comment (ybiquitous) [inline CHANGELOG.md:11]: [trivial] Please remove an extra space.

```suggestion
- **Cppcheck** Enable new recommended configuration and add `suppressions-list` option [#2474](https://github.com/sider/runners/pull/2474)
```

**Review by ybiquitous — APPROVED** [review]
LGTM 👍🏼

File: CHANGELOG.md (hunk 6 above)
Comment (sleeeto) [reply CHANGELOG.md]: Thanks, I fixed it.

**Pull request merged**